packages/mobile/src/exchange/ExchangeHomeScreen.tsx

```diff
@@ -1,126 +1,231 @@
 import BigNumber from 'bignumber.js'
-import * as React from 'react'
-import { WithTranslation } from 'react-i18next'
-import { StyleSheet, View } from 'react-native'
+import React, { useCallback, useEffect, useMemo, useRef } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import Animated from 'react-native-reanimated'
 import SafeAreaView from 'react-native-safe-area-view'
-import { connect } from 'react-redux'
+import { useDispatch } from 'react-redux'
 import CeloAnalytics from 'src/analytics/CeloAnalytics'
 import { CustomEventNames } from 'src/analytics/constants'
 import { fetchExchangeRate } from 'src/exchange/actions'
 import CeloGoldHistoryChart from 'src/exchange/CeloGoldHistoryChart'
 import CeloGoldOverview from 'src/exchange/CeloGoldOverview'
+import { useExchangeRate } from 'src/exchange/hooks'
+import { exchangeHistorySelector } from 'src/exchange/reducer'
 import { CURRENCY_ENUM } from 'src/geth/consts'
-import { Namespaces, withTranslation } from 'src/i18n'
+import { Namespaces } from 'src/i18n'
+import { LocalCurrencyCode } from 'src/localCurrency/consts'
+import { convertDollarsToLocalAmount } from 'src/localCurrency/convert'
+import { getLocalCurrencyExchangeRate } from 'src/localCurrency/selectors'
 import DrawerTopBar from 'src/navigator/DrawerTopBar'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
-import { RootState } from 'src/redux/reducers'
+import useSelector from 'src/redux/useSelector'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import TransactionsList from 'src/transactions/TransactionsList'
-import { getRateForMakerToken } from 'src/utils/currencyExchange'
+import { goldToDollarAmount } from 'src/utils/currencyExchange'
+import { getLocalCurrencyDisplayValue } from 'src/utils/formatting'
 
-interface StateProps {
-  exchangeRate: BigNumber
-  goldBalance: string | null
-  dollarBalance: string | null
-}
+type Props = StackScreenProps<StackParamList, Screens.ExchangeHomeScreen>
 
-interface DispatchProps {
-  fetchExchangeRate: typeof fetchExchangeRate
-}
+function ExchangeHomeScreen({ navigation }: Props) {
+  const scrollPosition = useRef(new Animated.Value(0)).current
+  const { t } = useTranslation(Namespaces.exchangeFlow9)
+  const dollarBalance = useSelector((state) => state.stableToken.balance)
+  const goldBalance = useSelector((state) => state.goldToken.balance)
+  const dispatch = useDispatch()
 
-type OwnProps = StackScreenProps<StackParamList, Screens.ExchangeHomeScreen>
+  // TODO: revert this back to `useLocalCurrencyCode()` when we have history data for cGDL to Local Currency.
+  const localCurrencyCode = null
+  const localExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const currentExchangeRate = useExchangeRate()
+  const goldToDollars = useCallback((amount) => goldToDollarAmount(amount, currentExchangeRate), [
```

Comment:
reading this again (https://kentcdodds.com/blog/usememo-and-usecallback), just wondering if the useCallback is worth it here or if it just adds unnecessary complexity

Reply:
I've moved unnecessary `useCallback` to the top of the component and defined them as `function() { ... }` instead of arrow function. I think, it has improved the readability. LMK WDYT?